lib/components/artworks/index.js

```diff
@@ -0,0 +1,224 @@
+    }
+  }
+
+  sectionedArtworks() {
+    const sectionedArtworks = [];
+    const sectionRatioSums = [];
+    for (let i = 0; i < this.props.sectionCount; i++) {
+      sectionedArtworks.push([]);
+      sectionRatioSums.push(0);
+    };
+
+    const artworks = this.state.artworks;
+    for (let i = 0; i < artworks.length; i++) {
+      const artwork = this.state.artworks[i];
+
+      // Find section with lowest *inverted* aspect ratio sum, which is the shortest column.
+      let lowestRatioSum = Number.MAX_VALUE; // Start higher, so we always find a
+      let sectionIndex = null;
+      for (let j = 0; j < sectionRatioSums.length; j++) {
+        const ratioSum = sectionRatioSums[j];
+        if (ratioSum < lowestRatioSum) {
+          sectionIndex = j;
+          lowestRatioSum = ratioSum;
+        }
+      }
+
+      const section = sectionedArtworks[sectionIndex];
+      section.push(artwork);
+
+      // Keep track of total section aspect ratio
+      const aspectRatio = artwork.image.aspect_ratio || 1; // Ensure we never divide by null/0
+      // Invert the aspect ratio so that a lower value means a shorter section.
+      sectionRatioSums[sectionIndex] += (1 / aspectRatio);
+    }
+
+    return sectionedArtworks;
+  }
+
+  renderSections() {
+    const spacerStyle = {
+      height: this.props.itemMargin,
+    };
+
+    const sectionedArtworks = this.sectionedArtworks();
+    const sections = [];
+    for (let i = 0; i < this.props.sectionCount; i++) {
+      const artworkComponents = [];
+      const artworks = sectionedArtworks[i];
+      for (let j = 0; j < artworks.length; j++) {
+        artworkComponents.push(<Artwork artwork={artworks[j]} key={'artwork-'+j} />);
+        // Setting a marginBottom on the artwork component didn’t work, so using a spacer view instead.
+        if (j < artworks.length-1) {
+          artworkComponents.push(<View style={spacerStyle} key={'spacer-'+j} accessibilityLabel='Spacer View' />);
+        }
+      }
+
+      const sectionSpecificStyle = {
+        width: this.state.sectionDimension,
+        marginRight: (i == this.props.sectionCount-1 ? 0 : this.props.sectionMargin),
+      };
+      sections.push(
+        <View style={[styles.section, sectionSpecificStyle]} key={i} accessibilityLabel={'Section ' + i}>
+          {artworkComponents}
+        </View>
+      );
+    }
+    return sections;
+  }
+
+  // Lifted pretty much straight from RN’s ListView.js
+  onScroll(event) {
+    const scrollProperties = event.nativeEvent;
+    const contentLength = scrollProperties.contentSize.height;
+    if (contentLength !== this._sentEndForContentLength) {
+      const offset = scrollProperties.contentOffset.y;
+      const visibleLength = scrollProperties.layoutMeasurement.height;
+      const distanceFromEnd = contentLength - visibleLength - offset;
+      if (distanceFromEnd < 100) {
+        this._sentEndForContentLength = contentLength;
+        this.fetchNextPage();
```

Comment:
Hmm, I now realise that I didn’t try to do anything special handling for the last page. Filing a ticket for that.